### PR TITLE
Add fashion design

### PR DESCRIPTION
**Pull request by andxu282**
### Summary 

This pull request is to add the Fashion Design Major. 

- [x] Add Fashion Design Major to index.ts and 

### Test Plan 
- attempt to fill out courses in the major
- double check for double counting
- test the bug found below

### CODE REVIEW

**Comment by jerrry1123** [comment]
Everything looks pretty good there's just a few bugs with AP courses that I caught. 

image
There's a bug involving AP Statistics where it should not count for either the Social Science or the Statistics distributions but it double counts it right now. 

image
The other thing is that AP courses are counting for the major's additional requirements even though I'm pretty sure they are not supposed to.

File: src/requirements/data/majors/fsad.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  'KCM',
+  'MQR',
+  'LA',
+  'CA',
+  'HA',
+];
+
+const fashionDesignNaturalScienceRequirements: readonly string[] = ['PBS', 'BIOLS-AG', 'BIONLS-AG'];
+
+const fashionDesignRequirements: readonly CollegeOrMajorRequirement[] = [
```
Comment (jerrry1123) [inline src/requirements/data/majors/fsad.ts:18]: see the comment i left in the main pr page for more details, but i found a few bugs relating to how AP courses count for these requirements, and there also might be additional bugs of a simliar nature that i did not catch.

**Review by jerrry1123 — CHANGES_REQUESTED** [review]

File: src/requirements/data/majors/fsad.ts (hunk 1 above)
Comment (einc) [reply src/requirements/data/majors/fsad.ts:18]: @benjamin-shen i think you saw this, but the ap classes have bugs 😭 it's not caused by this PR so I'm gonna add this bug to the notion to be tested before we push to prod.

as we talked off-line, maybe we potentially don't let ap classes automatically count for reqs, but limit it to user opt-in until we fix some of the bigger ap bugs??

the opt-in feature is gonna be added by andrew/will soon so at least there's a workaround (soon) for now

File: src/requirements/data/majors/fsad.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  {
+    name: 'Social Science',
```
Comment (einc) [inline src/requirements/data/majors/fsad.ts:20]: let's add "disallowTransferCredit: true" here as a temp fix for now bc AP stats is counting for this but it shouldn't

here's of an example of when we use it:
Screen Shot 2022-04-01 at 1 55 07 AM

File: src/requirements/data/majors/fsad.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  },
+  {
+    name: 'Statistics',
+    description:
+      'This fulfills the college distribution quantitative and analytical courses requirement. Must be taken at Cornell, AP Statistics is not accepted.',
+    source: 'https://courses.cornell.edu/preview_program.php?catoid=45&poid=23626',
+    checker: includesWithSubRequirements(['PAM 2100', 'AEM 2100', 'STSCI 2100', 'PSYCH 2500']),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
```
Comment (einc) [inline src/requirements/data/majors/fsad.ts:61]: also add "disallowTransferCredit: true" here to not let AP stats count

File: src/requirements/data/majors/fsad.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+            ifCodeMatch(catalogNbr, '6***'))
+        );
+      },
+    ],
+    fulfilledBy: 'courses',
+    perSlotMinCount: [2],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Fashion Design Additional Requirements',
```
Comment (einc) [inline src/requirements/data/majors/fsad.ts:204]: also add "disallowTransferCredit: true" here to not let AP stats count

File: src/requirements/data/majors/fsad.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+  },
+  {
+    name: 'Fashion Design Additional Courses',
+    description:
+      'Take any two FSAD courses at the 3000, 4000, or 6000 level. Course work from Fashion Design Core Courses cannot count here. FSAD 4000, 4010, 4020, 4030, 4990 cannot count here.',
+    source: 'https://courses.cornell.edu/preview_program.php?catoid=45&poid=23626',
+    checker: [
+      (course: Course): boolean => {
+        const { subject, catalogNbr } = course;
+        return (
```
Comment (einc) [inline src/requirements/data/majors/fsad.ts:191]: adding to check that "FSAD 4000, 4010, 4020, 4030, 4990 cannot count here

here's a similar example in CS:
Screen Shot 2022-04-01 at 1 54 00 AM
"

**Review by einc — CHANGES_REQUESTED** [review]
overall looks good, just left a few small changes to make and then ready to merge!!

File: src/requirements/data/majors/fsad.ts (hunk 1 above)
Comment (einc) [reply src/requirements/data/majors/fsad.ts:18]: we're temp not automatically applying ap credits (pr: https://github.com/cornell-dti/course-plan/pull/676#pullrequestreview-928913970), so this problem should be temp fixed rn!

**Review by einc — APPROVED** [review]

**Review by jerrry1123 — APPROVED** [review]

**Review by willespencer — APPROVED** [review]

**Pull request merged**